EducationalTestingService/rstfinder
# Improve readability and add new entry points.

### Changes in this PR

Closes #51 and closes #52 

#### Several readability changes to core code
- Add docstrings to all files/classes/functions/methods.
- Add author and organization information to all files.
- Resort imports at the top and make relative imports clearer.
- Change all format string to 3.6-style.
- Change all single quoted strings to double quotes
- Make inputs paths absolute in `tune_rst_parser.py` for easier interaction with SKLL.
- Remove backslashes in code and use parentheses.

#### Update requirements file and create conda version
- Update SKLL to v2.0 and unpin other requirements. We should really be using the latest version of SKLL but may be that can be done later. 
- Add scikits.bootstrap to `requirements.txt`
- Create separate `conda_requirements` file since `scikits-bootstrap` (conda-forge) != `scikits.bootstrap` (PyPI) 🤦 

#### Update tests
- Make tests runnable with `nose`.
- Fix and sort imports.
- Use` nose.tools` where appropriate.
- Add docstrings. 

#### Move all the stuff under `util` to `utils` under the main codebase. 

#### Update setup.py
- Use `find_packages()` instead of using package name.
 - Add new entry points for `visualize_rst_tree` and `compute_bootstrap_from_predictions` from the `utils` submodule.
- Add new entry point for `make_traindev_split` which was also a standalone script before.

### How to test

Currently, this can only be tested on a Linux machine and only on a ETS server because the `crf++` and `python-spar` conda packages are only available from our internal channel. 

1. Create a conda environment: `conda create --override-channels -c conda-forge -c ets -c https://nlp.research.ets.org/conda -p /scratch/nmadnani/parserdev36 python=3.6 nose crf++ --file conda_requirements.txt`.
2. Activate the environment: `conda activate parserdev36`. 
3. Install the package: `pip install -e .`
4. Run the tests: `nosetests -v`
5. Run all the steps in the README – they should all work; `make_traindev_split` which can now be called directly since it's an endpoints rather than using the separate script. 
6. Please make sure to check as many readability changes as possible to make sure they all look okay.
